Source/WTF/wtf/MallocSpan.h

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright (C) 2024 Apple Inc. All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY APPLE INC. AND ITS CONTRIBUTORS ``AS IS''
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+ * THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+ * PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL APPLE INC. OR ITS CONTRIBUTORS
+ * BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF
+ * THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#pragma once
+
+#include <utility>
+#include <wtf/FastMalloc.h>
+#include <wtf/Noncopyable.h>
+#include <wtf/StdLibExtras.h>
+
+// MallocSpan is a smart pointer class that wraps a std::span and calls fastFree in its destructor.
+
+namespace WTF {
+
+template<typename T, typename Malloc = FastMalloc> class MallocSpan {
+    WTF_MAKE_NONCOPYABLE(MallocSpan);
+public:
+    MallocSpan() = default;
+
+    MallocSpan(MallocSpan&& other)
+        : m_span(other.leakSpan())
+    {
+    }
+
+    ~MallocSpan()
+    {
+        Malloc::free(m_span.data());
```

Comment:
Hmmm... will default-constructed or moved-from MallocSpan end up freeing nullptr? I believe free of nullptr is OK in FastMalloc but not other mallocs.

Reply:
You've been working with C++ for so long that you've forgotten that free() is guaranteed to do nothing when you pass NULL. It's fine, at least according to the various standards bodies.

Reply:
MallocPtr was already relying on this behavior BTW: 
```
    ~MallocPtr()
    {
        Malloc::free(m_ptr);
    }
```